guide/interactions/replying-to-slash-commands.md

```diff
@@ -0,0 +1,256 @@
+
+```js {7}
+const wait = require('util').promisify(setTimeout);
+
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.defer(true);
+		await wait(4000);
+		await interaction.editReply('Pong!');
+	}
+});
+```
+
+Perfect, now you know how to reply to a slash command when you have to perform time intensive tasks!
+
+
+## Follow-ups
+
+Replying to slash commands is great and all, but what if you want to send multiple responses instead of just one? Follow-up messages got you covered, you can use `CommandInteraction#followUp()` to send multiple responses:
+
+::: warning
+After the initial response an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages.
+:::
+
+```js {6}
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply('Pong!');
+		await interaction.followUp('Pong again!');
+	}
+});
+```
+
+If you run this code you should end up having something that looks like this:
+
+<!--- vue-discord-message doesn't yet have support for inline replies/interactions/ephemeral messages -->
+<div is="discord-messages">
+	<discord-message profile="user">
+		/ping
+	</discord-message>
+	<discord-message profile="bot">
+		Pong!
+	</discord-message>
+	<discord-message profile="bot">
+		Pong again!
+	</discord-message>
+</div>
+
+Now you may want to send an ephemeral follow-up, to do so, just repeat the procedure as follows and also pass in `ephemeral: true` to the `InteractionReplyOptions`:
+
+```js {6}
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply('Pong!');
+		await interaction.followUp('Pong again!', { ephemeral: true });
+	}
+});
+```
+
+If you run this code you should end up having something that looks like this:
+
+<!--- vue-discord-message doesn't yet have support for inline replies/interactions/ephemeral messages -->
+<div is="discord-messages">
+	<discord-message profile="user">
+		/ping
+	</discord-message>
+	<discord-message profile="bot">
+		Pong!
+	</discord-message>
+	<discord-message profile="bot">
+		Pong again! (ephemeral)
+	</discord-message>
+</div>
+
+That's all, now you know everything there is to know on how to reply to slash commands! 
+
+::: tip
+Interaction responses run on top of webhooks thus, they inherit certain aspects from webhook messages:
+- they can contain up to 10 message embeds
+- they can use guild emojis globally
+- they can mask links in message content
+:::
+
+## Fetching and deleting responses
+
+Additionally to replying to a slash command you may also want to delete the initial reply. You can do this by using `CommandInteraction#deleteReply()` like this:
```

Comment:
Should mention that it's not possible to fetch or delete an ephemeral reply - edit only.

Reply:
Will do